src/cartesian_mesh.cc

```diff
@@ -48,4 +34,17 @@ double CartesianMesh::gradient(int /*i*/, double f_lo, double f_hi) const
     return (f_hi-f_lo)/step_; 
     }
 
+void CartesianMesh::validateBoundaryCondition(){
+    if (lower_bc_ != BoundaryType::periodic && upper_bc_ == BoundaryType::periodic){
+        throw std::invalid_argument("Lower boundary type not valid for periodic boundary condition");
+        }
+    else if (lower_bc_ == BoundaryType::periodic && upper_bc_ != BoundaryType::periodic){
+        throw std::invalid_argument("Upper boundary type not valid for periodic boundary condition");
+        }
```

Comment:
This is going to cause issues when slicing, since you can get `internal` + `periodic`. You should first check if either boundary condition is internal and `return` if true since you can't do any meaningful additional checks. Then, you can combine this into one check like:

```suggestion
    if ((lower_bc_ != BoundaryType::periodic && upper_bc_ == BoundaryType::periodic) ||
        (lower_bc_ == BoundaryType::periodic && upper_bc_ != BoundaryType::periodic))
        {
        throw std::invalid_argument("Both boundaries must be periodic if one is.");
        }
```